File: src/main/java/com/itsci/mju/maebanjumpen/entity/Hirer.java
package com.itsci.mju.maebanjumpen.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.EqualsAndHashCode;
import lombok.EqualsAndHashCode.Exclude;

import java.util.HashSet;
import java.util.Set;

@Entity
@DiscriminatorValue("hirer")
@AllArgsConstructor
@NoArgsConstructor
@Data
@JsonIgnoreProperties(ignoreUnknown = true)
@EqualsAndHashCode(callSuper = true, onlyExplicitlyIncluded = true)
@ToString(callSuper = true)
public class Hirer extends Member {

        @OneToMany(mappedBy = "hirer", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
        @ToString.Exclude
        @Exclude
        private Set<Hire> hires = new HashSet<>();
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/entity/Hirer.kt
package com.itsci.mju.maebanjumpen.entity

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.persistence.*

@Entity
@DiscriminatorValue("hirer")
@JsonIgnoreProperties(ignoreUnknown = true)
open class Hirer(
    @OneToMany(mappedBy = "hirer", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    open var hires: MutableSet<Hire> = mutableSetOf()
) : Member() {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Hirer) return false
        if (!super.equals(other)) return false
        return true
    }

    override fun hashCode(): Int = super.hashCode()
}

